src/main/java/org/mastodon/mamut/tomancak/lineage_registration/RootsPairing.java

```diff
@@ -0,0 +1,61 @@
+package org.mastodon.mamut.tomancak.lineage_registration;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.mastodon.collection.ObjectRefMap;
+import org.mastodon.collection.RefRefMap;
+import org.mastodon.collection.RefSet;
+import org.mastodon.collection.ref.ObjectRefHashMap;
+import org.mastodon.collection.ref.RefRefHashMap;
+import org.mastodon.mamut.model.ModelGraph;
+import org.mastodon.mamut.model.Spot;
+
+/**
+ * Pair the root spots in two ModelGraphs based on their label.
+ */
+public class RootsPairing
+{
+	/**
+	 * Pair the root nodes in two ModelGraphs based on their label.
+	 * A root node in graphA is paired with a root node in graphB
+	 * if it has the same label.
+	 * 
+	 * @param graphA graph A
+	 * @param graphB graph B   
+	 * @return a map, that maps root nodes in graph A to equally named
+	 * root nodes of graph B.
+	 */
+	static RefRefMap< Spot, Spot > pairRoots( ModelGraph graphA, ModelGraph graphB )
+	{
+		ObjectRefMap< String, Spot > rootsA = dividingRootsMap( graphA );
```

Comment:
aha, if they don't divide, we don't have to worry about them.. that was probably the intention
hmm... but wouldn't they still be useful for the registration -- finding the A->B transformation?

Reply:
I don't wont the "anterior", "posterior" spots or other solist spots be used during registration. I introduced this "only use dividing spots" as I though it could be a good criteria to filter these spots. Ideally the final plugin would also have an option to blacklist some spots. What are your thoughts about this?

Reply:
aha, got you now, I see
so there are basically three classes of roots:
- "solists" - a root that does not represent a living cell, and thus has no incoming/outgoing edges
- non-branching cells - a root that can be followed downstream (in time) to some leaf, but has no division along the way, gives rise to exactly one leaf
- branching - a root that.... has many leaves, builds a tree

My comment was targeting the non-branching ones as I have totally forgotten about the "solists" class. Btw, the solists from non-branching class may be distinguished by the length of the track/branch that they represent (could a be parameter of the plugin).

But, you are also right that both solists and non-branching are irrelevant for the tree matching. Since the matching is sensitive to the alignment, that is, to how accurate the registration-transformation is, maybe the irrelevant spots should not talk to it, should not influence it... and thus should be omitted, like your current code does. Also the algorithm would be less complicated (when it does not need to detect solists and such).

----------------
Hard to say... if user annotates/clicks initially a couple of spots, labels them, but then tracks only two trees, yet wants the two trees to have matched... the two nodes are not enough to do the alignment...

hmm, a parameter then?
but I don't want to complicate it (it's soooo easy to be coming up with requirements :-)

Reply:
@maarzt If do not change anything, I think, this would be fine. But then at least the javadoc should be more specific about what is done in this method (e.g. ignoring solists).

Reply:
I changed the method name to pairDividingRoots and also updated the javadoc.

Reply:
Okay.